GDB Stub not working

While the meson option `-Dgdb-stub=true` causes an extra file to be built, it never defines the GDB_STUB variable, causing support for it to not completely be enabled.

EDIT: A `--enable-gdb-stub` build with autotools doesn't work either, due to variable scoping. Even after fixing this, the `--arm9stub` option isn't recognized in the command-line, despite being present in the `--help`. What happened to break this so badly? This was all working not _that_ long ago...

Also adding to the issue, if you manually add `-DGDB_STUB` to both cpp_args and c_args on `meson configure`, it fails to compile because the gdb handles are inside the `static void common_gtk_main(GApplication *app, gpointer user_data)` scope, therefore, out of scope for the `static void Teardown()` function on the gtk backend:

> [159/163] Compiling C++ object gtk/desmume.p/main.cpp.o
> FAILED: gtk/desmume.p/main.cpp.o
> c++ -Igtk/desmume.p -Igtk -I../desmume/src/frontend/posix/gtk -I../desmume/src -I../desmume/src/libretro-common/include -I../desmume/src/frontend -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/SDL -I/usr/include/AL -I/usr/include/soundtouch -I/usr/include/gtk-3.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/gio-unix-2.0 -I/usr/include/atk-1.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/at-spi-2.0 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -std=c++14 -g -DEXPERIMENTAL_WIFI_COMM -DHAVE_JIT -DHOST_64 -DHAVE_GL_GLX -DHAVE_LIBSOUNDTOUCH -DGDB_STUB -pthread -D_GNU_SOURCE=1 -D_REENTRANT -MD -MQ gtk/desmume.p/main.cpp.o -MF gtk/desmume.p/main.cpp.o.d -o gtk/desmume.p/main.cpp.o -c ../desmume/src/frontend/posix/gtk/main.cpp
> ../desmume/src/frontend/posix/gtk/main.cpp: In function ‘void Teardown()’:
> ../desmume/src/frontend/posix/gtk/main.cpp:4437:22: error: ‘arm9_gdb_stub’ was not declared in this scope
>  4437 |     destroyStub_gdb( arm9_gdb_stub);
>       |                      ^~~~~~~~~~~~~
> ../desmume/src/frontend/posix/gtk/main.cpp:4440:22: error: ‘arm7_gdb_stub’ was not declared in this scope
>  4440 |     destroyStub_gdb( arm7_gdb_stub);
>       |

--- a/desmume/src/frontend/posix/gtk/main.cpp
+++ b/desmume/src/frontend/posix/gtk/main.cpp
@@ -104,6 +104,11 @@ VideoFilter* video;
 
 desmume::config::Config config;
 
+#ifdef GDB_STUB
+gdbstub_handle_t arm9_gdb_stub = NULL;
+gdbstub_handle_t arm7_gdb_stub = NULL;
+#endif
+
 enum {
     MAIN_BG_0 = 0,
     MAIN_BG_1,
@@ -3864,10 +3869,7 @@ common_gtk_main(GApplication *app, gpointer user_data)
      */
 #ifdef GDB_STUB
     gdbstub_mutex_init();
-    
-    gdbstub_handle_t arm9_gdb_stub = NULL;
-    gdbstub_handle_t arm7_gdb_stub = NULL;
-    
+
     if ( my_config->arm9_gdb_port > 0) {
         arm9_gdb_stub = createStub_gdb( my_config->arm9_gdb_port,
                                          &NDS_ARM9,

--- a/desmume/src/frontend/posix/meson.build
+++ b/desmume/src/frontend/posix/meson.build
@@ -219,6 +219,7 @@ endif
 #endif
 
 if get_option('gdb-stub')
+  add_global_arguments('-DGDB_STUB', language: ['c', 'cpp'])
   libdesmume_src += [
     '../../gdbstub/gdbstub.cpp',
   ]

